scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd

```diff
@@ -0,0 +1,410 @@
+---
+title: "Setup: Introduction Module"
+author: "Stephanie Spielman for the Data Lab"
+date: "`r Sys.Date()`"
+output: 
+  html_notebook:
+    toc: true
+    toc_depth: 3
+    toc_float: true
+    number_sections: false
+---
+
+## Setup
+
+```{r setup, include=FALSE}
+set.seed(2022)
+
+library(magrittr)
+
+library(SingleCellExperiment)
+library(SingleR)
+library(bluster) 
+```
+
+
+### Download Data
+
+To get the raw data:
+```{sh eval = FALSE}
+NAME=10k_PBMCs_TotalSeq_B_3p_raw_feature_bc_matrix.tar.gz
+wget https://cf.10xgenomics.com/samples/cell-exp/6.0.0/10k_PBMCs_TotalSeq_B_3p/$NAME
+tar -xf $NAME
+# This gets unpacked into `raw_feature_bc_matrix`
+
+
+# For posterity, first attempt was with MALT, but it had wildly low ADT counts
+#NAME=malt_10k_protein_v3_raw_feature_bc_matrix.tar.gz
+#wget https://cf.10xgenomics.com/samples/cell-exp/3.0.0/malt_10k_protein_v3/$NAME
+```
+
+## Loading Data
+
+```{r}
+pbmc_10x_dir <- "raw_feature_bc_matrix"
+# for now..
+mito_file <- "https://raw.githubusercontent.com/AlexsLemonade/scpca-downstream-analyses/main/reference-files/Homo_sapiens.GRCh38.104.mitogenes.txt"
+
+
+
+pbmc_sce_raw <- DropletUtils::read10xCounts(pbmc_10x_dir)
+mito_genes <- readr::read_tsv(mito_file, 
+                              col_names="gene") %>%
+  # only keep relevant mitochondrial genes
+  dplyr::filter(gene %in% rownames(pbmc_sce_raw)) %>%
+  dplyr::pull(gene)
+
+# let's have a quick look at the raw SCE and genes:
+pbmc_sce_raw
+mito_genes
```

Comment:
Minor, but for teaching I would separate these chunks and put the `pbmc_sce_raw` call to print out the SCE directly after reading in the data. And then in a new chunk do the mito file.

Reply:
And on that note, we might want to reorganize in general, so that the creation of the `mito_genes` variable actually happens before `addPerCellQC` instead of all the way up here.

Reply:
Yes, I expect we will split the up much more for teaching!